src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -10,42 +15,191 @@ namespace Infrastructure.Common
     // refer to members within their test class or its base classes.
     public class ConditionalWcfTest
     {
-        private static bool IsConditionalTestEnabled(string testCategoryName)
+        private static ConcurrentDictionary<string, bool> _evaluatedConditions = new ConcurrentDictionary<string, bool>();
+
+        // Returns 'true' or 'false' for the given condition name.
+        // If this is the first call and 'detectFunc' is specified, it will be
+        // called to detect whether the condition is true or false.  If not
+        // specified, it will default to false. The value is cached after first call.
+        private static bool GetConditionValue(string conditionName, Func<bool> detectFunc = null)
```

Comment:
The existence of `GetConditonValue(...)` and `GetConditionValueFromTestProperties(...)` are very confusing. I recommend us to conclude [this design discussion](https://github.com/dotnet/wcf/pull/1026#issuecomment-211263175) and merge these two functions to be one that always follows the same searching order.